control.lua

```diff
@@ -198,25 +211,29 @@ local function destroy_marked_items(player, pos, nano_ammo, deconstructors)
     if entity.to_be_deconstructed(player.force) and (nano_ammo.valid and nano_ammo.valid_for_read) and not table_find(global.queued, find_match, entity) then
       if deconstructors then
         local item_list = {}
-
         entity.surface.create_entity{name="nano-cloud-small-deconstructors", position=entity.position, force="neutral"}
         nano_ammo.drain_ammo(1)
 
         --Get all the damn items and clear the inventories
         for item_name, count in pairs(get_all_items_inside(entity)) do
```

Comment:
My error, but since you are here. we can optimize this by wrapping it in an entity.has_items_inside() if check